scapy/base_classes.py

```diff
@@ -102,9 +99,7 @@ def __iter__(self):
                     for a in range(*self.parsed[0]):
                         yield "%i.%i.%i.%i" % (a,b,c,d)
     def choice(self):
-        ip = []
-        for v in self.parsed:
-            ip.append(str(random.randint(v[0],v[1]-1)))
+        ip = (str(random.randint(v[0],v[1]-1)) for v in self.parsed)
         return ".".join(ip) 
```

Comment:
You can even unify those lines

Reply:
Yep. I know, but did not want to change sources to deeply :)
Of course I will change it in next commit.

Reply:
Thanks :)

Reply:
Done